lib/wallet/src/Cardano/Wallet/Transaction.hs

```diff
@@ -507,6 +534,26 @@ emptyTokenMapWithScripts = TokenMapWithScripts
     , txScripts = Map.empty
     }
 
+data WitnessCount = WitnessCount
+    { verificationKey :: Word8
+    , scripts :: [AnyScript]
```

Comment:
well, witness set of tx, when there are native script engaged, there NEEDS to be value of `Script KeyHash`. Otherwise, ie., in the absence of script - ITS STRUCTURE, ledger would not able to verify if verification key witness are correct. Example to illustrate the case:
```
all [verKey1, verKey2]   produces scripthash1
all [verKey1, verKey2, from slot1]   produces scripthash2
``` 
and scripthash1 is diffrent from scripthash2
LEDGER needs to have structure of native script present, script hash is not enough as it is result of one way function!
To sum up ledger NEEDS to have (1) verKey1, verKey2 AND (2) witnesses from signing duals of them wit1, wit2 AND (3) script structure to say if wits are correct.

I think it is useful addition to have knowledge how native script looks like in a given tx and to be obtained that during decode transaction call. Nevertheless, extending plutus structure and adding references is a natural NEXT step what we should do in next PRs